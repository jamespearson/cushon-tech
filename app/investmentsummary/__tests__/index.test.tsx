import { render, screen, fireEvent } from '@testing-library/react-native'
import { router } from 'expo-router'
import InvestmentSummaryScreen from '..'

import fixtureFund from '../../fixtures/fund.json'

let mockData: any = fixtureFund
let mockLoading = false
jest.mock('../../hooks/useFundDetails', () => {
  return () => ({
    data: mockData,
    isLoading: mockLoading,
    error: undefined
  })
})

describe('Investment Success Screen', () => {
  describe('when loading', () => {
    beforeEach(() => {
      mockData = undefined
      mockLoading = true
    })

    it('should render as expected', () => {
      render(<InvestmentSummaryScreen />)

      expect(screen.toJSON()).toMatchSnapshot()
    })

    it('should show a loading indicator', () => {
      render(<InvestmentSummaryScreen />)

      expect(screen.getByTestId('activity-indicator')).toBeTruthy()
    })
  })

  describe('when the data has loaded', () => {
    beforeEach(() => {
      mockData = fixtureFund.fund
      mockLoading = false
    })

    it('should render as expected', () => {
      render(<InvestmentSummaryScreen />)

      expect(screen.toJSON()).toMatchSnapshot()
    })

    it('should display fund information', () => {
      render(<InvestmentSummaryScreen />)
      expect(screen.getByText('CushonMix medium risk/return ISA')).toBeTruthy()
    })

    // NOTE: The useRef used to check if a field has been touched is not working in the test environment.

    // describe('when the user enters a negative amount', () => {
    //   it('should show an error', async () => {
    //     render(<InvestmentSummaryScreen />)

    //     fireEvent.changeText(screen.getByTestId('amount-input'), '-1')

    //     expect(screen.getByText('Please enter a valid amount')).toBeTruthy()
    //   })
    // })

    // describe('when the user enters a valid amount', () => {
    //   it('should not show an error', async () => {
    //     render(<InvestmentSummaryScreen />)

    //     fireEvent.changeText(screen.getByTestId('amount-input'), '1')

    //     expect(screen.getByText('Please enter a valid amount')).toBeFalsy()
    //   })
    // })
  })
})
