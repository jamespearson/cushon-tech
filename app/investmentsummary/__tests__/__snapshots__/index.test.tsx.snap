// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Investment Success Screen when loading should render as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size="large"
    testID="activity-indicator"
  />
</View>
`;

exports[`Investment Success Screen when the data has loaded should render as expected 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "gap": 24,
        "padding": 24,
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        {
          "flex": 1,
        }
      }
      waitFor={
        [
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <View
          style={
            {
              "gap": 24,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "MuseoSansRounded500",
                "fontSize": 32,
                "lineHeight": 42,
                "textAlign": "center",
              }
            }
          >
            CushonMix medium risk/return ISA
          </Text>
          <Text
            style={
              {
                "fontFamily": "MuseoSansRounded700",
                "fontSize": 18,
                "lineHeight": 24,
              }
            }
          >
            This CushonMix portfolio spreads your money across a specially selected range of top-rated funds from our comparison tables.
          </Text>
          <Text
            style={
              {
                "fontFamily": "MuseoSansRounded300",
                "fontSize": 14,
                "lineHeight": 22,
              }
            }
          >
            Spreading your money allows each individual fund to be slightly riskier, which helps boost projected returns, whilst still controlling the overall risk of the investment.

This means the CushonMix offers higher projected returns than alternative investment strategies with the same risk profile. Find out more about the CushonMix portfolios.

Remember, you can set up as many separate ISA pots as you like with Cushon and they all sit in one main account behind the scenes. This is helpful if you want to ring-fence different pots for different savings goals, or to spread your money across different risk profiles.
          </Text>
          <Image
            source={1}
            style={
              {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "borderColor": "red",
          "borderRadius": 8,
          "borderWidth": 1,
          "gap": 12,
          "padding": 24,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "MuseoSansRounded700",
            "fontSize": 18,
            "lineHeight": 24,
          }
        }
      >
        Invest now
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 12,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          Â£
        </Text>
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="25000"
          style={
            {
              "alignSelf": "stretch",
              "backgroundColor": "white",
              "borderColor": "rgba(0,0,0,0.4)",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "fontFamily": "MuseoSansRounded300",
              "fontSize": 16,
              "paddingHorizontal": 24,
              "paddingVertical": 24,
            }
          }
          testID="amount-input"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "stretch",
              "backgroundColor": "#df2f89",
              "borderColor": "#df2f89",
              "borderRadius": 8,
              "borderWidth": 1,
              "padding": 24,
            }
          }
        >
          <View
            style={{}}
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "MuseoSansRounded700",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              GO
            </Text>
          </View>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;
